TACC/Core-Portal
# task/FP-1560 Use _common Sidebar

## Overview

The goal was to integrate the previously created common Sidebar component into other components that had sidebars but were using a different implementation. There were changes made to the Sidebar component as well as changes made to the structure of some components  to accommodate this new sidebar.


## Related

* [FP-1560](https://jira.tacc.utexas.edu/browse/FP-1560)

## Changes
Sidebar component changes: 
- Added a `backgroundColor` prop. By default the background for the sidebar is gray. This can be overridden by passing in a different color through the prop.
- Removed `addItems` prop and added `addItemsBefore` and `addItemsAfter` props. This ensures that if we want to add other items to the sidebar (such as buttons, headings etc.) we can add them either before or after the sidebar nav items. Prior to this, we could only add items after the nav items. 

Other component changes: 
- Changes were made to how the file and data were structured in the components to match the data the Sidebar component expects. 


## Testing

1. Go to the dashboard 
2. Click the links in the sidebar and ensure they work properly and navigate to the proper page
3. Test the sidebar in the following pages ensuring proper navigation: 
    - Data Files
    - Allocations
    - History
4. Search something from the top right searchbar and ensure sidebar on the SiteSearch page works as expected.

## UI

- No major UI changes except for the SiteSearch page where the sidebar has a border now.

Before: 

<img width="1792" alt="Screen Shot 2022-11-03 at 1 51 39 PM" src="https://user-images.githubusercontent.com/23081932/199815601-6bb3f4b3-8e69-4d6e-a7aa-8b6e62eabf49.png">

After: 

<img width="1792" alt="Screen Shot 2022-11-03 at 1 52 03 PM" src="https://user-images.githubusercontent.com/23081932/199815669-253e5a6a-0df9-4b27-908d-96a45f0191ff.png">


## Notes
- The [Applications](https://github.com/TACC/Core-Portal/blob/main/client/src/components/Applications/AppBrowser/AppBrowser.jsx) component has not been switched yet to use the common sidebar. The implementation is a bit more complicated for this component since it doesn't behave the same as the other components. I will be creating a separate task to add the common sidebar to the Applications component. 
- Once this is approved and tested there will be cleanup needed to remove unused components and css.
